client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -215,6 +215,30 @@ private Checkpoint completeCheckpoint(String checkpointName) {
     public void resetReaderGroup(ReaderGroupConfig config) {
         log.info("Reset ReaderGroup {} to {}", getGroupName(), config);
         synchronizer.fetchUpdates();
+
+        //If null passed to ReaderGroupConfig then reset the reader group to last completed checkpoint.
+        if (config == null) {
+            val latestCheckpointConfig = synchronizer.getState().getConfig();
+            Optional<Map<Stream, Map<Segment, Long>>> lastCheckPointPositions =
+                    synchronizer.getState().getPositionsForLastCompletedCheckpoint();
+
+            Map<Stream, StreamCut> streamCuts = new HashMap<>();
+            if (lastCheckPointPositions.isPresent()) {
+                for (Entry<Stream, Map<Segment, Long>> streamPosition : lastCheckPointPositions.get().entrySet()) {
+                    streamCuts.put(streamPosition.getKey(), new StreamCutImpl(streamPosition.getKey(), streamPosition.getValue()));
+                }
+
+            } else {
+                log.info("Reset reader group to last completed checkpoint is not successful as there is no checkpoint available, so resetting to start of the stream ");
+                Set<String> streamNames = getStreamNames();
+                for (String stream : streamNames) {
+                    streamCuts.put(Stream.of(stream), StreamCut.UNBOUNDED);
```

Comment:
Yes, got your point. My initial assumption was when there is no checkpoint completed and there a call for resetReaderGroup(null) then reset the reader group to start of stream but If we consider the case where application requested read for given streamcut then it make sense to read from start of streamcut rather than start of stream itself. Will do the necessary changes.